sopht/utils/restart_sim.py

```diff
@@ -0,0 +1,46 @@
+from typing import Type, Union
+import os
+import elastica as ea
+from sopht.simulator.flow.navier_stokes_flow_simulators import (
+    UnboundedNavierStokesFlowSimulator3D,
+    UnboundedNavierStokesFlowSimulator2D,
+)
+from sopht.simulator.flow.passive_transport_flow_simulators import (
+    PassiveTransportFlowSimulator,
+)
+from sopht.utils.io import IO
+
+
+def restart_simulation(
+    flow_sim: Union[
+        UnboundedNavierStokesFlowSimulator2D,
+        UnboundedNavierStokesFlowSimulator3D,
+        PassiveTransportFlowSimulator,
+    ],
+    restart_simulator: Type[ea.BaseSystemCollection],
+    io: IO,
+    rod_io: IO,
+    forcing_io: IO,
+    restart_dir: str,
```

Comment:
Minor comment, but it would be nice if we can load flow data from a custom directory as well. Probably not needed because flow data is always saved in the current directory.

Reply:
Yes, I think if we want to do this we should also change the save directory for sopht. Because the other way user needs to copy the file into another directory.

Reply:
Okay let's leave it as it is for now. We may make things more robust later.